From bf6f6f5988aa6a998796d414915aee8056afee20 Mon Sep 17 00:00:00 2001
From: David S. Wang <dsw@cloudera.com>
Date: Wed, 27 Mar 2013 16:55:06 -0700
Subject: [PATCH 44/96] HBASE-6655 thrift.TestThriftServerCmdLine takes too much time: 733.972 sec

Reason: Test Failure
Author: Nicolas Liochon
Ref: CDH-10954
(cherry picked from commit bd5a339d6f600c6b50481c9428202bc8d20ab9be)
---
 .../hadoop/hbase/thrift/TestThriftServer.java      |    4 ++++
 .../hbase/thrift/TestThriftServerCmdLine.java      |    9 ++++++---
 2 files changed, 10 insertions(+), 3 deletions(-)

diff --git a/src/test/java/org/apache/hadoop/hbase/thrift/TestThriftServer.java b/src/test/java/org/apache/hadoop/hbase/thrift/TestThriftServer.java
index 81409c4..90f00ac 100644
--- a/src/test/java/org/apache/hadoop/hbase/thrift/TestThriftServer.java
+++ b/src/test/java/org/apache/hadoop/hbase/thrift/TestThriftServer.java
@@ -189,6 +189,10 @@ public class TestThriftServer {
     assertEquals(handler.getTableNames().size(), 2);
   }
 
+  public static void checkTableList(Hbase.Iface handler) throws Exception {
+    assertTrue(handler.getTableNames().contains(tableAname));
+  }
+
   public static void dropTestTables(Hbase.Iface handler) throws Exception {
     handler.disableTable(tableBname);
     assertFalse(handler.isTableEnabled(tableBname));
diff --git a/src/test/java/org/apache/hadoop/hbase/thrift/TestThriftServerCmdLine.java b/src/test/java/org/apache/hadoop/hbase/thrift/TestThriftServerCmdLine.java
index 516f321..7e2c3da 100644
--- a/src/test/java/org/apache/hadoop/hbase/thrift/TestThriftServerCmdLine.java
+++ b/src/test/java/org/apache/hadoop/hbase/thrift/TestThriftServerCmdLine.java
@@ -88,8 +88,8 @@ public class TestThriftServerCmdLine {
             continue;
           }
           for (boolean specifyCompact : new boolean[] {false, true}) {
-            parameters.add(new Object[]{implType, new Boolean(specifyFramed),
-                new Boolean(specifyBindIP), new Boolean(specifyCompact)});
+            parameters.add(new Object[]{implType, specifyFramed,
+                specifyBindIP, specifyCompact});
           }
         }
       }
@@ -164,7 +164,10 @@ public class TestThriftServerCmdLine {
 
     thriftServer = new ThriftServer(TEST_UTIL.getConfiguration());
     startCmdLineThread(args.toArray(new String[0]));
-    Threads.sleepWithoutInterrupt(2000);
+
+    while ( thriftServer.serverRunner == null || thriftServer.serverRunner.tserver == null ){
+      Thread.sleep(1);
+    }
 
     Class<? extends TServer> expectedClass = implType != null ?
         implType.serverClass : TBoundedThreadPoolServer.class;
-- 
1.7.0.4

